### ISSUE TITLE
Bug: Slice parameters are tokenized as part of the function itself

### ISSUE THREAD

**Issue by Duckapple** [ionide/ionide-fsgrammar#182]
**Describe the bug**
Slice parameters are tokenized as part of the function itself

**To Reproduce**
Steps to reproduce the behaviour:
1. Insert the following test into a text file (the other numbers in the screenshot are to demonstrate the color they should have):
```fs
let test1 (word: char list) = List.rev word[0 .. 1]
```

**Expected behaviour**
I would expect the numbers to still be highlighted as numbers, though I think parsing them as expressions is the correct move.

**Screenshots**
![image](https://user-images.githubusercontent.com/39386945/215521315-8d3ba219-b677-4276-b7e3-57f5c1e0aefe.png)

**Environment (please complete the following information):**
 - OS: Linux (6.1.6-arch1-1)
 - Ionide version: v7.4.0
 - VSCode version: 1.74.3
 - dotnet SDK version: 7.0.102
 - mono / .Net Framework version: (didn't know how to check / probably not relevant?)

**Additional context**
I would fix this myself, but I don't have the expertise with TextMate grammars I'd probably need.
It also seems like GitHub gets this part right with its grammar.

**Comment by MangelMaxime** [comment]
Hello,

Looking at the inspector I think the problem is not in the fsgrammar but in the semantic highlighter. The code responsible for the semantic highlight is hosted in https://github.com/fsharp/FsAutoComplete and can sometimes need a fix in https://github.com/dotnet/fsharp.

<img width="656" alt="image" src="https://user-images.githubusercontent.com/4760796/215544964-3edec83b-9ed4-464e-a8ad-b3d2e8f5a138.png">

@baronfel Am I right to say the problem is in the semantic highlighter ?

**Comment by baronfel** [comment]
yeah, that's right - anything in the `semantic token type` is FSAC, textmate tokens are this repo.

FSAC is pretty much a straight pass-through of the semantic highlighting from the FSharp.Compiler.Service library - so it's likely that any real fix would end up in dotnet/fsharp eventually.
